src/System.Reflection/tests/AssemblyNameTests.cs

```diff
@@ -187,16 +187,20 @@ public void Flags_CurrentlyExecutingAssembly()
         public void FullName(string name, string expectedName)
         {
             AssemblyName assemblyName = new AssemblyName(name);
-            
+
             string extended = $"{expectedName}, Culture=neutral, PublicKeyToken=null";
-            Assert.True(assemblyName.FullName == expectedName || assemblyName.FullName == extended);
+            Assert.StartsWith(expectedName, assemblyName.FullName);
+            if (assemblyName.FullName.Length > expectedName.Length)
+            {
+                Assert.Equal(extended, assemblyName.FullName);
+            }
```

Comment:
I prefer the old way, as it's much easier for me to understand at a glance what it's checking for.